core/services/pipeline/variables.go

```diff
@@ -85,10 +85,28 @@ func (vars Vars) Set(dotID string, value interface{}) {
 	dotID = strings.TrimSpace(dotID)
 	if len(dotID) == 0 {
 		panic(ErrVarsRoot)
-	} else if strings.IndexByte(dotID, keypathSeparator[0]) >= 0 {
-		panic("cannot set a nested key of a pipeline.Vars")
 	}
-	vars.vars[dotID] = value
+	keys := strings.Split(dotID, string(keypathSeparator))
+	if len(keys) > 2 {
+		panic("cannot set a keypath more than 2 deep")
+	}
+	parent := vars.vars
+	for i, key := range keys {
+		if i == len(keys)-1 {
+			parent[key] = value
+			break
+		}
+		switch child := parent[key].(type) {
+		case map[string]interface{}:
+			parent = child
+		case nil:
+			newChild := make(map[string]interface{})
+			parent[key] = newChild
+			parent = newChild
+		default:
+			panic("cannot set a key of a non map[string]interface{} type")
```

Comment:
Maybe just throw an error instead of panicking in runtime?

Reply:
wanted to keep it consistent with previous functionality, which panics